### PR TITLE
LPS-116779 SF

### PR DESCRIPTION
**Pull request by carolmariaabb**
Related issue: https://issues.liferay.com/browse/LPS-116779

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+		DDMFormInstance ddmFormInstance =
+			DDMFormInstanceTestUtil.addDDMFormInstance(
+				ddmForm, _group, TestPropsValues.getUserId());
+
+		DDMFormValues ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(
+			ddmForm);
+
+		ddmFormValues.addDDMFormFieldValue(
+			DDMFormValuesTestUtil.createDDMFormFieldValue(
+				"text", new UnlocalizedValue("Approved Text")));
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java:96]: @marcosapmf, the first three tests have this code in common then can we extract it to a separated method in order to re-use and avoid code duplication?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+		_upgradeDDMFormInstanceReport.upgrade();
+
+		ddmFormInstanceReport = _getDDMFormInstanceReport(
+			ddmFormInstance.getFormInstanceId());
+
+		JSONObject actualDataJSONObject = _jsonFactory.createJSONObject(
+			ddmFormInstanceReport.getData());
+
+		Assert.assertEquals(
+			expectedDataJSONObject.toString(), actualDataJSONObject.toString());
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java:137]: @marcosapmf, same as above.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+
+		return ddmForm;
+	}
+
+	private void _createDDMFormInstanceRecords(DDMFormInstance ddmFormInstance)
+		throws Exception {
+
+		DDMForm ddmForm = _createDDMForm();
+
+		int qtd = 0;
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java]: @marcosapmf, is `qtd` an acronym for `quantity`? Would be better we use something like `size` ou `length`. 😉

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option2", LocaleUtil.US, "option2");
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option3", LocaleUtil.US, "option3");
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option4", LocaleUtil.US, "option4");
+
+		ddmForm.addDDMFormField(checkboxMultipleDDMFormField);
+
+		DDMFormField gridFormField = DDMFormTestUtil.createDDMFormField(
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java:405]: @marcosapmf, please rename to `gridDDMFormField`.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java [hunk 5]
```diff
@@ -103,10 +103,10 @@ public void testUpgradeDDMFormInstanceReportWhenFormRecordHasDraft()
 				ddmFormInstance.getFormInstanceId(), ddmFormValues,
 				serviceContext);
 
-		DDMFormValues updateDDMFormValues =
+		DDMFormValues newDDMFormValues =
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java:106]: Good job, @carolmariaabb! 😄

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java [hunk 6]
```diff
@@ -315,7 +315,7 @@ public void testUpgradeWithExistingDDMFormInstanceReports()
 	}
 
 	@Test
-	public void testUpgradeWithNoFormInstanceReports() throws Exception {
+	public void testUpgradeWithNoDDMFormInstanceReports() throws Exception {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java:318]: Nice, @carolmariaabb!

**Pull request closed without merging**